OpenBLAS build broken on AIX

/opt/IBM/openxlC/17.1.2/bin/ibm-clang_r -c -O2 -DSMALL_MATRIX_OPT -DGEMM_GEMV_FORWARD -DSBGEMM_GEMV_FORWARD -DMAX_STACK_ALLOC=2048 -fopenmp -DHAVE_P10_SUPPORT -Wall -DF_INTERFACE_IBM -fPIC -DDYNAMIC_ARCH -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=16 -DMAX_PARALLEL_NUMBER=1 -DBUILD_BFLOAT16 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.34.dev\" -DUSE_OPENMP -fopenmp -fno-integrated-as -m64 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=lsame -DASMFNAME=lsame_ -DNAME=lsame_ -DCNAME=lsame -DCHAR_NAME=\"lsame_\" -DCHAR_CNAME=\"lsame\" -DNO_AFFINITY -DTS= -I.. -DF_INTERFACE ../kernel/power/lsame.S -o lsame.o
Assembler:
/tmp/lsame-fa9d22.s: line 62: 1252-016 The specified opcode or pseudo-op is not valid.
        Use supported instructions or pseudo-ops only.
.ibm-clang: error: assembler command failed with exit code 1 (use -v to see invocation)
Assembler:
/tmp/cabs-3215e4.s: line 62: 1252-016 The specified opcode or pseudo-op is not valid.
        Use supported instructions or pseudo-ops only.
.ibm-clang: error: assembler command failed with exit code 1 (use -v to see invocation)
Assembler:
/tmp/cabs-153920.s: line 62: 1252-016 The specified opcode or pseudo-op is not valid.
        Use supported instructions or pseudo-ops only.
.ibm-clang: error: assembler command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:188: lsame.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:176: scabs1.o] Error 1
make[1]: *** [Makefile:179: dcabs1.o] Error 1
make[1]: Leaving directory '/home/buildusr/jenkins/workspace/OpenBLAS-openxl/OpenBLAS/kernel'
make: *** [Makefile:208: libs] Error 1

Checking the assembly file shows error pointing to "PROLOGUE_EXPORT". 

Looks like the recent change from #5577 caused this regression on AIX.

--- a/common_alpha.h
+++ b/common_alpha.h
@@ -145,6 +145,16 @@ static __inline int blas_quickdivide(unsigned int x, unsigned int y){
 	.ent REALNAME; \
 REALNAME:
 
+#define PROLOGUE_EXPORT \
+	.arch ev6; \
+	.set noat; \
+	.set noreorder; \
+.text; \
+	.align 5; \
+	.globl REALNAME; \
+	.ent REALNAME; \
+REALNAME:
+
 #ifdef PROFILE
 #define PROFCODE \
 	ldgp	$gp, 0($27); \

--- a/common_ia64.h
+++ b/common_ia64.h
@@ -371,6 +371,13 @@ static __inline int blas_quickdivide(unsigned int x, unsigned int y){
 	.proc REALNAME; \
 REALNAME:
 
+#define PROLOGUE_EXPORT \
+	.explicit; \
+	.text; \
+	.align 128; \
+	.global REALNAME; \
+	.proc REALNAME; \
+REALNAME:
 
 #ifdef PROFILE
 #define PROFCODE \

--- a/common_power.h
+++ b/common_power.h
@@ -668,6 +668,17 @@ REALNAME:;\
 	.csect .text[PR],5;\
 .REALNAME:
 
+#define PROLOGUE_EXPORT \
+	.machine "any";\
+	.toc;\
+	.globl .REALNAME;\
+	.globl REALNAME;\
+	.csect REALNAME[DS],3;\
+REALNAME:;\
+	.long .REALNAME, TOC[tc0], 0;\
+	.csect .text[PR],5;\
+.REALNAME:
+
 #define EPILOGUE \
 _section_.text:;\
 	.csect .data[RW],4;\
@@ -676,6 +687,17 @@ _section_.text:;\
 #else
 
 #define PROLOGUE \
+	.machine "any";\
+	.toc;\
+	.globl .REALNAME;\
+	.globl REALNAME;\
+	.csect REALNAME[DS],3;\
+REALNAME:;\
+	.llong .REALNAME, TOC[tc0], 0;\
+	.csect .text[PR], 5;\
+.REALNAME:
+
+#define PROLOGUE_EXPORT \
 	.machine "any";\
 	.toc;\
 	.globl .REALNAME;\
@@ -707,8 +729,28 @@ _section_.text:;\
 	.globl REALNAME
 REALNAME:
 	.endmacro
+
+	.macro PROLOGUE_EXPORT
+	.section __TEXT,__text,regular,pure_instructions
+	.section __TEXT,__picsymbolstub1,symbol_stubs,pure_instructions,32
+	.machine ppc
+	.text
+	.align 4
+	.globl REALNAME
+REALNAME:
+	.endmacro
 #else
 	.macro PROLOGUE
+	.section __TEXT,__text,regular,pure_instructions
+	.section __TEXT,__picsymbolstub1,symbol_stubs,pure_instructions,32
+	.machine ppc64
+	.text
+	.align 4
+	.globl REALNAME
+REALNAME:
+	.endmacro
+
+	.macro PROLOGUE_EXPORT
 	.section __TEXT,__text,regular,pure_instructions
 	.section __TEXT,__picsymbolstub1,symbol_stubs,pure_instructions,32
 	.machine ppc64

--- a/common_sparc.h
+++ b/common_sparc.h
@@ -211,6 +211,14 @@ static __inline int blas_quickdivide(blasint x, blasint y){
 	.proc	07; \
 REALNAME:;
 
+#define PROLOGUE_EXPORT \
+	.section	".text"; \
+	.align 32; \
+	.global REALNAME;\
+	.type	REALNAME, #function; \
+	.proc	07; \
+REALNAME:;
+
 #if defined(__linux__) && defined(__ELF__)
 #define GNUSTACK .section .note.GNU-stack,"",@progbits
 #else

